src/main/java/org/sonatype/repository/vgo/internal/util/VgoDataAccess.java

```diff
@@ -112,10 +115,29 @@ public Asset findAsset(final StorageTx tx, final Bucket bucket, final String ass
    *
    * @return blob content
    */
-  public Content saveAsset(final StorageTx tx,
-                           final Asset asset,
-                           final Supplier<InputStream> contentSupplier,
-                           final Payload payload) throws IOException
+  @TransactionalStoreBlob
+  public Content doCreateOrSaveAsset(final Repository repository,
+                                     final String assetPath,
+                                     final VgoAssetKind assetKind,
+                                     final TempBlob tempBlob,
+                                     final Payload payload) throws IOException
+  {
+    StorageTx tx = UnitOfWork.currentTx();
+    Bucket bucket = tx.findBucket(repository);
+
+    Asset asset = findAsset(tx, bucket, assetPath);
```

Comment:
This code block is duplicated. Extract it to findOrCreateAsset and just have a ternary operator : ```component != null ? component : format```

Reply:
I dont think this is suitable, it would mean having to pass component and format for each place it is used and also if/else on the tx.createAsset because they are different types. Making it less readable IMO